### PR TITLE
Bump examples to 2.0.0.

### PR DESCRIPTION
**Pull request by anthonyshew**
### Description

Bumping examples to 2.0.0. 🥳 

## Testing Instructions

CI

### CODE REVIEW

File: examples/with-docker/package.json [hunk 1]
```diff
@@ -14,13 +14,12 @@
...
   },
-  "dependencies": {},
   "devDependencies": {
     "@repo/eslint-config": "*",
     "prettier": "^3.2.5",
-    "turbo": "^1.12.4"
+    "turbo": "2.0.0-canary.2"
   },
-  "packageManager": "yarn@1.22.19",
+  "packageManager": "yarn@3.6.3",
```
Comment (chris-olszewski) [inline examples/with-docker/package.json:22]: Was the Yarn 1 -> Yarn 3 migration intentional?

File: examples/with-npm/package.json [hunk 2]
```diff
@@ -14,10 +14,10 @@
   "devDependencies": {
     "@repo/eslint-config": "*",
     "prettier": "^3.2.5",
-    "turbo": "^1.12.4"
+    "turbo": "^2.0.0-canary.2"
   },
-  "packageManager": "npm@8.19.4",
+  "packageManager": "npm@10.5.0",
```
Comment (chris-olszewski) [inline examples/with-npm/package.json:19]: Again, flagging NPM version bump

File: examples/with-yarn/package.json [hunk 3]
```diff
@@ -14,10 +14,10 @@
   "devDependencies": {
     "@repo/eslint-config": "*",
     "prettier": "^3.2.5",
-    "turbo": "^1.12.4"
+    "turbo": "2.0.0-canary.2"
   },
-  "packageManager": "yarn@1.22.19",
+  "packageManager": "yarn@3.6.3",
```
Comment (chris-olszewski) [inline examples/with-yarn/package.json:19]: ⚠️ Yarn 1 -> Yarn 3 move

**Review by chris-olszewski — APPROVED** [review]
LFG

Flagged a few package manager bumps that I think were unintentional.

File: examples/non-monorepo/package.json [hunk 4]
```diff
@@ -19,10 +19,10 @@
     "@types/react-dom": "^18.2.19",
     "eslint": "^8.57.0",
     "eslint-config-next": "^14.1.1",
-    "turbo": "^1.12.4",
+    "turbo": "^2.0.0-canary.2",
     "typescript": "^5.3.2"
   },
-  "packageManager": "npm@8.19.4",
+  "packageManager": "npm@10.5.0",
```
[the remark this answers is not in the document]
Comment (anthonyshew) [reply examples/non-monorepo/package.json:25]: Good with this one. Good flag.

File: examples/with-npm/package.json (hunk 2 above)
Comment (anthonyshew) [reply examples/with-npm/package.json:19]: Good with it!

**Review by chris-olszewski — APPROVED** [review]

**Pull request merged**